### PR TITLE
Add new definition for jkiten-react-date-range

### PR DESCRIPTION
**Pull request by Jkiten**
Please fill in this template.

- [v] Use a meaningful title for the pull request. Include the name of the package modified.
- [v] Test the change in your own code. (Compile and run.)
- [v] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [v] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [v] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [v] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [v] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [v] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [v] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [v] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [v] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [v] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by Jkiten** [comment]
> Generally you should publish types in your package instead of using DT. Is there a reason you're doing this PR?

hello.
I needed a function to use a package called react-date-range, so I added the necessary parts and published a library that only I use. However, when I was doing next.js with typescript, I got an error while deploying, so I needed @types, so I did pr.
Could you please let me know if there is any other workaround or something I need to fix? thank you

File: types/jkiten-react-date-range/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,344 @@
+// Type definitions for jkiten-react-date-range 1.4
+// Project: https://github.com/OnionFarm-Corp/jkiten-react-date-range
```
Comment (peterblazejewicz) [inline types/jkiten-react-date-range/index.d.ts:2]: I'me getting 404 here

File: types/jkiten-react-date-range/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+// Definitions by: Jkiten <https://github.com/OnionFarm-Corp>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
```
Comment (peterblazejewicz) [inline types/jkiten-react-date-range/index.d.ts:5]:
```suggestion
```
that has no longer a meaning here, TS4.3 is the lowest one, so just remove probably

**Comment by peterblazejewicz** [comment]
> Could you please let me know if there is any other workaround or something I need to fix?

usually one would add `index.d.ts` alongside the `index.js` of source module, make sure it's listed as part of NPM publish step in `package.json` and just republish package to NPM:
https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package

**Comment by Jkiten** [comment]
hello
Thanks for your help.
As the person in charge said, I added and distributed the x.d.ts file, and the library ran without any problems.
Now I think I can close the pr I sent :)
I'll send you a request next time I need it again.
thank you so much for caring

**Pull request closed without merging**